src/main/groovy/lang/ObjectRange.java

```diff
@@ -518,25 +496,41 @@ private T peek() {
     }
 
     /**
-     * Increments by one
+     * Increments by step size
      *
      * @param value the value to increment
-     * @return the incremented value
+     * @param step  the value to increment by
+     * @return the incremented value or null, if there isn't any
      */
     @SuppressWarnings("unchecked")
-    private T increment(T value) {
-        return (T) InvokerHelper.invokeMethod(value, "next", null);
+    private T increment(T value, int step) {
+        for (int i = 0; i < step; i++) {
+            final T next = (T) InvokerHelper.invokeMethod(value, "next", null);
+            if (!compareGreaterThan(next, value)) /* e.g. `next` of the last element */ {
+                return null;
+            }
+            value = next;
+        }
+        return value;
     }
 
     /**
-     * Decrements by one
+     * Decrements by step size
      *
      * @param value the value to decrement
-     * @return the decremented value
+     * @param step  the value to decrement by
+     * @return the decremented value or null, if there isn't any
      */
     @SuppressWarnings("unchecked")
-    private T decrement(T value) {
```

Comment:
I'd like to see this rebased. That would help me understand whether we have broken binary compatibility.